packages/core/src/util/resolvers.ts

```diff
@@ -155,35 +155,25 @@ export const resolveSchema = (
   // schemas around
   if (resultSchema !== undefined && resultSchema.$ref !== undefined) {
     try {
-      return retrieveResolvableSchema(schema, resultSchema.$ref);
+      const resolvableSchema = resolveSchema(schema, resultSchema.$ref);
+      if (resolvableSchema !== undefined){
+        return resolvableSchema;
+      } 
+      if (rootSchema){
+        return resolveSchema(rootSchema, resultSchema.$ref);
+      }
+      return schema;
     } catch (e) {
```

Comment:
Try/catch should be removed as it's no longer necessary. Lines 157-171 can be replaced with `return resolveSchema(rootSchema, resultSchema.$ref)`